packages/tables/src/elements/Row.spec.tsx

```diff
@@ -176,4 +176,35 @@ describe('Row', () => {
       expect(getByTestId('row')).toHaveStyleRule('height', '64px');
     });
   });
+
+  describe('ReadOnly', () => {
+    it('removes tabIndex while in read-only mode', () => {
+      const { getByTestId } = render(
+        <Table isReadOnly>
+          <Body>
+            <Row data-test-id="row" />
+          </Body>
+        </Table>
+      );
+
+      expect(getByTestId('row')).not.toHaveAttribute('tabindex');
+    });
+
+    it('removes :hover styling while in read-only mode', () => {
+      const { getByTestId } = render(
+        <Table isReadOnly>
+          <Body>
+            <Row data-test-id="row" />
+          </Body>
+        </Table>
+      );
+
+      expect(getByTestId('row')).toHaveStyleRule('border-bottom-color', undefined, {
+        modifier: '&:hover'
+      });
+      expect(getByTestId('row')).toHaveStyleRule('background-color', undefined, {
+        modifier: '&:hover'
+      });
```

Comment:
@jzempel @hzhu I'm unable to get the `modifier` values here to pick up `:hover` styling 😢  It's finding `:focus` fine though.

Do you all have any ideas on getting this to work?

Reply:
I've had challenges with these, as well. I think the `toHaveStyleRule` tests work best when they directly target the `StyledRow` type components, rather than the high-order wrapper.

Reply:
I think instead of picking up `undefined`, the `.not` API can be used to ensure the border styling is not applied on hover. One suggestion is something along the lines of:

```tsx
it.only('does not show :hover styling while in read-only mode', () => {
  const { getByTestId } = render(
    <Table isReadOnly>
      <Body>
        <Row data-test-id="row" />
      </Body>
    </Table>
  );

  const row = getByTestId('row');

  userEvent.hover(row);

  expect(row).not.toHaveStyleRule(
    'border-bottom-color',
    getColor('primaryHue', 200, DEFAULT_THEME),
    {
      modifier: '&:hover'
    }
  );
});
```

Reply:
And the other test case would be:

```tsx
it.only('shows :hover styling while in read-only mode', () => {
  const { getByTestId } = render(
    <Table>
      <Body>
        <Row data-test-id="row" />
      </Body>
    </Table>
  );

  const row = getByTestId('row');

  userEvent.hover(row);

  expect(row).toHaveStyleRule(
    'border-bottom-color',
    getColor('primaryHue', 200, DEFAULT_THEME),
    {
      modifier: '&:hover'
    }
  );
});
```

Reply:
When I tried running something similar to this locally `toHaveStyleRule` starts to produce errors that it can't understand the identifier provided. I also tried wrapping it in a `css` block but that didn't help either.

Reply:
These were running locally for me 🤔 .